### PR TITLE
peer+server: ensure the peer is always able to quit even mid msgStream application 

### LINKED ISSUE #1789: go1.11: golang.org/x/crypto: relocation target runtime.support_avx2 not defined
### Background

```
[0:06:19] i:maurycy:~/go/src/github.com/lightningnetwork/lnd> make clean && make
Cleaning source.
rm -f ./lnd ./lncli
rm -f -r ./vendor
Compiling dependencies.
dep ensure -v
(1/42) Wrote gopkg.in/macaroon.v2@bed2a428da6e56d950bed5b41fcbae3141e5b0d0
(2/42) Wrote github.com/lightningnetwork/lightning-onion@ac4d9da8f1d67c95f1fafdc65e1a4902d6f5a940
(3/42) Wrote github.com/miekg/dns@79bfde677fa81ff8d27c4330c35bda075d360641
(4/42) Wrote github.com/btcsuite/go-socks@master
(5/42) Wrote github.com/juju/loggo@master
(6/42) Wrote github.com/aead/siphash@master
(7/42) Wrote github.com/lightninglabs/gozmq@master
(8/42) Wrote github.com/kkdai/bstream@f391b8402d23024e7c0f624b31267a89998fca95
(9/42) Wrote github.com/btcsuite/btclog@84c8d2346e9fc8c7b947e243b9c24e6df9fd206a
(10/42) Wrote github.com/btcsuite/btcutil@ab6388e0c60ae4834a1f57511e20c17b5f78be4b
(11/42) Wrote github.com/lightninglabs/neutrino@166fe699d5964581d24e6bfff0aa329cfb6a8bc9
(12/42) Wrote github.com/btcsuite/fastsha256@master
(13/42) Wrote github.com/jrick/logrotate@a93b200c26cbae3bb09dd0dc2c7c7fe1468a034a
(14/42) Wrote github.com/NebulousLabs/fastrand@master
(15/42) Wrote github.com/ltcsuite/ltcd@cdab10132e8c6e4a3ffd112dba54791946d28906
(16/42) Wrote github.com/rogpeppe/fastuuid@master
(17/42) Wrote github.com/btcsuite/btcwallet@7b84dc25a61634c450d21ec7f47d5e916eb88fdb
(18/42) Wrote github.com/tv42/zbase32@501572607d0273fc75b3b261fa4904d63f6ffa0e
(19/42) Wrote github.com/btcsuite/websocket@master
(20/42) Wrote github.com/coreos/bbolt@4f5275f4ebbf6fe7cb772de987fa96ee674460a7
(21/42) Wrote github.com/urfave/cli@1efa31f08b9333f1bd4882d61f9d668a70cd902e
(22/42) Wrote github.com/btcsuite/golangcrypto@master
(23/42) Wrote github.com/davecgh/go-spew@8991bc29aa16c548c550c7ff78260e27b9ab7c73
(24/42) Wrote github.com/go-errors/errors@a6af135bd4e28680facf08a3d206b454abc877a4
(25/42) Wrote github.com/btcsuite/btcd@79e00513b1011888b1e675157ab89f527f901cae
(26/42) Wrote github.com/golang/protobuf@bbd03ef6da3a115852eaf24c8a1c46aeb39aa175
(27/42) Wrote github.com/jackpal/gateway@v1.0.4
(28/42) Wrote github.com/grpc-ecosystem/grpc-gateway@f2862b476edcef83412c7af8687c9cd8e4097c0f
(29/42) Wrote github.com/jackpal/go-nat-pmp@28a68d0c24adce1da43f8df6a57340909ecd7fdd
(30/42) Wrote github.com/jessevdk/go-flags@f88afde2fa19a30cf50ba4b05b3d13bc6bae3079
(31/42) Wrote gopkg.in/errgo.v1@v1
(32/42) Wrote google.golang.org/grpc@b3ddf786825de56a4178401b7e174ee332173b66
(33/42) Wrote github.com/aead/chacha20@master
(34/42) Wrote golang.org/x/crypto@49796115aa4b964c318aad4f3084fdb41e9aa067
(35/42) Wrote github.com/NebulousLabs/go-upnp@29b680b06c82d044ebea91bf3069038eb562df2a
(36/42) Wrote gopkg.in/macaroon-bakery.v2@94012773d2874a067572bd16d7d11ae02968b47b
(37/42) Wrote git.schwanenlied.me/yawning/bsaes.git@master
(38/42) Wrote github.com/Yawning/aez@4dad034d9db2caec23fb8f69b9160ae16f8d46a3
(39/42) Wrote google.golang.org/genproto@df60624c1e9b9d2973e889c7a1cff73155da81c4
(40/42) Wrote golang.org/x/sys@master
(41/42) Wrote golang.org/x/net@ae89d30ce0c63142b652837da33d782e2b0a9b25
(42/42) Wrote golang.org/x/text@v0.3.0
 Building debug lnd and lncli.
go build -v -tags=debug -o lnd-debug -ldflags "-X main.Commit=5cf911a762bdbbe9e6536026d450dbf3d9a36503" github.com/lightningnetwork/lnd
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/internal
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/naming
github.com/lightningnetwork/lnd/vendor/github.com/grpc-ecosystem/grpc-gateway/utilities
github.com/lightningnetwork/lnd/vendor/github.com/golang/protobuf/proto
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/codes
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/grpclog
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/metadata
github.com/lightningnetwork/lnd/lnwire
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/connectivity
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/credentials
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/stats
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/peer
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/tap
github.com/lightningnetwork/lnd/brontide
github.com/lightningnetwork/lnd/channeldb
github.com/lightningnetwork/lnd/vendor/github.com/golang/protobuf/ptypes/struct
github.com/lightningnetwork/lnd/vendor/github.com/grpc-ecosystem/grpc-gateway/runtime/internal
github.com/lightningnetwork/lnd/vendor/github.com/golang/protobuf/ptypes/any
github.com/lightningnetwork/lnd/vendor/google.golang.org/genproto/googleapis/rpc/status
github.com/lightningnetwork/lnd/vendor/github.com/golang/protobuf/protoc-gen-go/descriptor
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/status
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/grpclb/grpc_lb_v1
github.com/lightningnetwork/lnd/vendor/github.com/golang/protobuf/jsonpb
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/transport
github.com/lightningnetwork/lnd/vendor/google.golang.org/genproto/googleapis/api/annotations
github.com/lightningnetwork/lnd/vendor/gopkg.in/macaroon-bakery.v2/bakery/internal/macaroonpb
github.com/lightningnetwork/lnd/vendor/gopkg.in/macaroon-bakery.v2/bakery
github.com/lightningnetwork/lnd/vendor/github.com/grpc-ecosystem/grpc-gateway/runtime
github.com/lightningnetwork/lnd/autopilot
github.com/lightningnetwork/lnd/lnwallet
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc
github.com/lightningnetwork/lnd/macaroons
github.com/lightningnetwork/lnd/lnrpc
github.com/lightningnetwork/lnd/lnpeer
github.com/lightningnetwork/lnd/contractcourt
github.com/lightningnetwork/lnd/walletunlocker
github.com/lightningnetwork/lnd/htlcswitch
github.com/lightningnetwork/lnd/routing
github.com/lightningnetwork/lnd/zpay32
github.com/lightningnetwork/lnd/discovery
github.com/lightningnetwork/lnd
# github.com/lightningnetwork/lnd
github.com/lightningnetwork/lnd/vendor/golang.org/x/crypto/blake2b.supportsAVX2: relocation target runtime.support_avx2 not defined
github.com/lightningnetwork/lnd/vendor/golang.org/x/crypto/blake2b.supportsAVX: relocation target runtime.support_avx not defined
make: *** [build] Error 2
[0:06:44] i:maurycy:~/go/src/github.com/lightningnetwork/lnd> git rev-parse --verify HEAD
5cf911a762bdbbe9e6536026d450dbf3d9a36503
[0:07:02] i:maurycy:~/go/src/github.com/lightningnetwork/lnd> go version
go version go1.11 darwin/amd64
```

### Your environment

* `lnd version 0.4.2-beta commit=5cf911a762bdbbe9e6536026d450dbf3d9a36503`
* `Darwin i 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64`
* `btcd version 0.12.0-beta`
* `Bitcoin Core Daemon version v0.16.1.0-gdac5d68fc6cf`
* `go version go1.11 darwin/amd64`

### Steps to reproduce

`make`

Please see the attached typescript log.

### Expected behaviour

Success

### Actual behaviour

`Error`

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we patch a few holes in the "all goroutines need to be able to gracefully exit" rule we have within the codebase. Before this commit, it was possible that the chan messages stream for the link, and the gossiper for a peer would block indefinitely. We fix this by threading through the quit channel for the peer, and also allowing the `waitForChannelOpen` method to take a caller specified quit channel. 

Finally, we also ensure that we no longer attempt to enable to disable private channels due to inactivity. This is a minor optimization, as we don't need to sign signatures, and also we avoid a db update. 

Fixes #1789

### CODE REVIEW

File: peer.go [hunk 1]
```diff
@@ -431,14 +431,19 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) error {
 		p.activeChannels[chanID] = lnChan
 		p.activeChanMtx.Unlock()
 
-		activeChans = append(activeChans, *chanPoint)
+		// Only if the channel is publicdo we need to collect it for
```
Comment (cfromknecht) [inline peer.go]: public do

File: peer.go [hunk 2]
```diff
@@ -738,6 +743,7 @@ func (ms *msgStream) Stop() {
 func (ms *msgStream) msgConsumer() {
 	defer ms.wg.Done()
 	defer peerLog.Tracef(ms.stopMsg)
```
Comment (cfromknecht) [inline peer.go]: not a blocker, but this should prob be deferred after printing the start message

File: peer.go [hunk 3]
```diff
@@ -738,6 +743,7 @@ func (ms *msgStream) Stop() {
...
+	defer atomic.StoreInt32(&ms.streamShutdown, 1)
```
Comment (cfromknecht) [inline peer.go]: 👌

File: peer.go [hunk 4]
```diff
@@ -878,6 +891,14 @@ func newDiscMsgStream(p *peer) *msgStream {
 //
 // NOTE: This method MUST be run as a goroutine.
 func (p *peer) readHandler() {
+	chanMsgStreams := make(map[lnwire.ChannelID]*msgStream)
+
+	defer p.wg.Done()
+	defer func() {
+		for _, chanStream := range chanMsgStreams {
+			chanStream.Stop()
```
Comment (cfromknecht) [inline peer.go]: Why not defer right after each is started?

File: peer.go (hunk 1 above)
Comment (Roasbeef) [reply peer.go]: Fixed.

File: peer.go (hunk 4 above)
Comment (Roasbeef) [reply peer.go]: Fixed.

**Comment by Roasbeef** [comment]
Pushed out a new version, fixing some issues, and also allowing `lnd` to build with go `1.11`.

**Pull request merged**